src/gtfs/mod.rs

```diff
@@ -256,22 +256,30 @@ struct Shape {
     sequence: u32,
 }
 
-fn read<H>(
-    file_handler: &mut H,
-    config_path: Option<impl AsRef<Path>>,
-    prefix: Option<String>,
-    on_demand_transport: bool,
-) -> Result<Model>
+///parameters consolidation
+#[derive(Clone)]
+pub struct Configuration<P: AsRef<Path>> {
+    /// path to configuration file
+    pub config_path: Option<P>,
+    /// used to prefix objects
+    pub prefix: Option<String>,
+    /// stop time precision management
+    pub on_demand_transport: bool,
+    /// on demand transport comment template
+    pub on_demand_transport_comment: Option<String>,
+}
+
+fn read<H>(file_handler: &mut H, configuration: Configuration<impl AsRef<Path>>) -> Result<Model>
```

Comment:
Interesting, any specific reason that made you choose
```
fn read<H>(file_handler: &mut H, configuration: Configuration<impl AsRef<Path>>) -> Result<Model>
```

over something like
```
fn read<H, P: AsRef<Path>>(file_handler: &mut H, configuration: Configuration<P>) -> Result<Model>
```

Reply:
apparently it's the same with a small difference https://stackoverflow.com/a/47515540

Reply:
Excellent link, thank you.